src/compat/eme/generate_key_request.ts

```diff
@@ -139,19 +147,17 @@ export function patchInitData(initData : Uint8Array) : Uint8Array {
  */
 export default function generateKeyRequest(
   session: MediaKeySession|ICustomMediaKeySession,
-  initData: Uint8Array,
-  initDataType: string|undefined
+  initializationData : IInitializationDataInfo
 ) : Observable<unknown> {
   return observableDefer(() => {
     log.debug("Compat: Calling generateRequest on the MediaKeySession");
     let patchedInit : Uint8Array;
     try {
-      patchedInit = patchInitData(initData);
+      patchedInit = patchInitData(initializationData.data);
     } catch (_e) {
-      patchedInit = initData;
+      patchedInit = initializationData.data;
     }
-    return castToObservable(session.generateRequest(initDataType == null ? "" :
-                                                                           initDataType,
-                                                    patchedInit));
+    const initDataType = initializationData.type ?? "";
```

Comment:
I think that we shouldn't call generateRequest with an empty string. The EME spec says :
`If initDataType is the empty string, return a promise rejected with a newly created TypeError.`

Reply:
I think that `initDataType` is `undefined` only on older browser which does not respect the last EME spec version.

Also `session` here might also be an `ICustomMediaKeySession` which is mostly implemented in-JS.
For example, the `WebkitMediaKeySession` defined in `src/compat/eme/custom_media_keys/webkit_media_keys.ts` does not use the `initDataType` in its `generateRequest` implementation.

Reply:
In the other hand, I agree that using an empty string here is an ugly work-around.

Maybe we should be more explicit and just communicate `undefined` when we don't have one.
Though I'm afraid of possible regressions here.

Reply:
Maybe using "cenc" as a default when unknown?

Reply:
Maybe using "cenc" and retrying if it fails?

Reply:
But in this case, retrying with what?
Maybe it makes more sense to try the empty string first, see if we're encountering a TypeError in a rejected Promise and in that case, try "cenc"?

Reply:
Ok

Reply:
done